cirq/pasqal/pasqal_device.py

```diff
@@ -11,52 +11,45 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Any, Dict, FrozenSet, Iterable
-from numpy import sqrt
+from typing import FrozenSet, Sequence
 
 import cirq
-
-from cirq.pasqal import ThreeDGridQubit
+from cirq.ops import NamedQubit
 
 
 @cirq.value.value_equality
 class PasqalDevice(cirq.devices.Device):
-    """A Pasqal Device with qubits placed on a 3D grid."""
+    """A generic Pasqal device."""
 
-    def __init__(self, control_radius: float,
-                 qubits: Iterable[ThreeDGridQubit]) -> None:
-        """ Initializes the description of the device.
+    def __init__(self, qubits: Sequence[cirq.ops.Qid]) -> None:
+        """Initializes a device with some qubits.
 
         Args:
-            control_radius: the maximum distance between qubits for a controlled
-                gate. Distance is measured in units of the indices passed into
-                the qubit constructor.
-            qubits: Qubits on the device, identified by their x, y, z position.
-                Must be of type ThreeDGridQubit.
-
+            qubits (NamedQubit): Qubits on the device, exclusively unrelated to
+                a physical position.
         Raises:
-            ValueError: if the wrong qubit type is provided or if invalid
-                parameter is provided for control_radius. """
-        us = cirq.value.Duration(micros=1)
-        ms = cirq.value.Duration(millis=1)
-
-        self._measurement_duration = 5 * ms
-        self._gate_duration = 2 * us
-        self._max_parallel_z = 2
-        self._max_parallel_xy = 2
-        self._max_parallel_c = 10
-        self._max_parallel_t = 1
+            TypeError: if the wrong qubit type is provided.
+        """
 
         for q in qubits:
-            if not isinstance(q, ThreeDGridQubit):
+            if not isinstance(q, self.supported_qubit_type):
                 raise TypeError('Unsupported qubit type: {!r}'.format(q))
```

Comment:
To make the error message more helpful, you could list the supported qubit types.